src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -64,31 +64,48 @@ public bool RemoveCachedSemanticModel(SyntaxTree tree)
 
             internal DeclarationAnalysisData GetOrComputeDeclarationAnalysisData(
                 SyntaxReference declaration,
-                Func<Func<DeclarationAnalysisData>, DeclarationAnalysisData> computeDeclarationAnalysisData,
+                Func<Func<DeclarationAnalysisDataBuilder>, DeclarationAnalysisDataBuilder> computeDeclarationAnalysisData,
                 bool cacheAnalysisData)
             {
-                if (!cacheAnalysisData)
+                DeclarationAnalysisDataBuilder dataBuilder = null;
```

Comment:
this is a super strange pattern for pooling and retrieving objects.  it seems very error prone.